### PR TITLE
Publisher trusted algos

### LINKED ISSUE #96: Checksum algo checks on provider

### LINKED ISSUE #97: Secure algorithms
Provider should check publisherTrustedAlgorithms from the assets ddos.
See EPIC for more details

### PR DESCRIPTION
**Pull request by calina-c**
Fixes #97.

### CODE REVIEW

**Comment by calina-c** [comment]
Issue blocked to enable testing from Ocean.JS (done) and Ocean.py (provider)

File: ocean_provider/validation/algo.py [hunk 1]
```diff
@@ -283,68 +283,78 @@ def validate(self):
...
+            did_to_trusted_algo_dict = {}
             self.error = (
                 "Some algos in the publisherTrustedAlgorithms don't have a did."
             )
-            return False
 
-        for trusted_algorithm in trusted_algorithms:
-            algo = get_asset_from_metadatastore(
-                get_metadata_url(), trusted_algorithm["did"]
+        if not self.error:
```
Comment (calina-c) [inline ocean_provider/validation/algo.py:310]: no need for this if. whenever self.error is set, return False. Then you can remove the if and continue the flow of testing with its contents

File: ocean_provider/validation/algo.py [hunk 2]
```diff
@@ -283,68 +283,78 @@ def validate(self):
...
+                self.error = f"containerSectionChecksum for algorithm with did {algo_ddo.did} does not match"
 
-        return True
+        return bool(self.error)
+
+    def validate_algo(self):
+        """Validates algorithm details that allow the algo dict to be built."""
+        algorithm_meta = self.data.get("algorithmMeta")
+        algorithm_did = self.data.get("algorithmDid")
+        if not algorithm_did and not algorithm_meta:
```
Comment (calina-c) [inline ocean_provider/validation/algo.py:343]: same here. The whole point with the validator is that these imbricated complex ifs are not necessary. Can you please return False whenever you set an error and redo this part? I'll re-review after.

**Comment by calina-c** [comment]
LGTM. :D

**Review by alexcos20 — APPROVED** [review]

**Pull request merged**